libs/dyn/value.go

```diff
@@ -44,8 +60,8 @@ func NewValue(v any, loc Location) Value {
 	}
 }
 
-// WithLocation returns a new Value with its location set to the given value.
-func (v Value) WithLocation(loc Location) Value {
+// WithLocations returns a new Value with its location set to the given value.
+func (v Value) WithLocations(loc []Location) Value {
```

Comment:
A slice is a pointer. An append may not copy if the slice has enough capacity. This means it is possible for 2 separate `append`'s to interfere with one another. To prevent this, you can store a copy of the incoming slice.

Also see #1295.

Reply:
Good catch. This would likely have surfaced as a bug in the future. Changed this and `NewValue` to create new copy of the input slices.